aio_kavenegar/client.py

```diff
@@ -0,0 +1,182 @@
+import json
+
+from typing import Literal
+
+import httpx
+
+from aio_kavenegar.exceptions import APIException, HTTPException
+from aio_kavenegar.types import KavenegarResponse
+
+
+# Default requests timeout in seconds.
+DEFAULT_TIMEOUT: int = 10
+DEFAULT_HOST: str = "api.kavenegar.com"
+DEFAULT_HEADERS: dict = {
+    "Accept": "application/json",
+    "Content-Type": "application/x-www-form-urlencoded",
+    "charset": "utf-8",
+}
+
+
+class AIOKavenegarAPI:
+    """
+    https://kavenegar.com/rest.html
+    """
+
+    def __init__(
+        self,
+        apikey: str,
+        timeout: int = DEFAULT_TIMEOUT,
+        host: str = DEFAULT_HOST,
+        headers: dict = DEFAULT_HEADERS,
+        proxies: dict | None = None,
+        secure: bool = True,
+    ) -> None:
+        """
+        :param str apikey: Kavengera API Key
+        :param int timeout: request timeout, default is 10
+        :param str host: Kavenegar API host, default is `api.kavenegar.com`
+        :param dict headers: headers used when requesting Kavenegar resources, default:
+            {
+                "Accept": "application/json",
+                "Content-Type": "application/x-www-form-urlencoded",
+                "charset": "utf-8",
+            }
+        :param dict proxies: Dictionary mapping protocol to the URL of the proxy:
+            {
+                'http': 'http://192.168.1.10:3128',
+                'https': 'http://192.168.1.10:3129',
+            }
+        """
+        self.version: str = "v1"
+        self.host: str = host
+        self.apikey: str = apikey
+        self.apikey_mask: str = f"{apikey[:2]}********{apikey[-2:]}"
+        self.timeout: int = timeout
+        self.headers: dict = headers
+        if not proxies:
+            self.mounts: dict = None
+        else:
+            self.mounts: dict = {}
+            if http := proxies.get("http"):
+                self.mounts.update({"http://": httpx.HTTPTransport(proxy=http)})
+            if https := proxies.get("https"):
+                self.mounts.update({"https://": httpx.HTTPTransport(proxy=https)})
+        self.secure = secure
+
+    def __repr__(self) -> str:
+        return "kavenegar.AIOKavenegarAPI({!r})".format(self.apikey_mask)
+
+    def __str__(self) -> str:
+        return "kavenegar.AIOKavenegarAPI({!s})".format(self.apikey_mask)
+
+    def _pars_params_to_json(self, params: dict) -> dict:
+        """
+        Kavenegar bug, the api server expects the parameters in a JSON-like array format,
+        but the requests library form-encode each key-value pair
+
+        Params (dict):
+        { sender: ["30002626", "30002627", "30002727", ], }
+
+        request behavior:
+        sender=30002626&sender=30002627&sender=30002727
+
+        Server expectation:
+        sender=["30002626","30002627","30002727"]
```

Comment:
I don't think it's a bug. It's just their API's behavior and their API doc examples clearly show it. IMO there's nothing wrong with it.

Reply:
Original library mentions it as a bug :)